#! /bin/sh /usr/share/dpatch/dpatch-run
## 21_atiusb.dpatch by Mario Limonciello <superm1@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make ATI Remotes have their own menu item and add snapstream support (LP: #140060)

@DPATCH@
diff -urNad lirc-0.8.3~pre1~/doc/lirc.hwdb lirc-0.8.3~pre1/doc/lirc.hwdb
--- lirc-0.8.3~pre1~/doc/lirc.hwdb	2007-10-30 15:48:16.000000000 -0400
+++ lirc-0.8.3~pre1/doc/lirc.hwdb	2007-10-30 15:48:16.000000000 -0400
@@ -113,8 +113,9 @@
 ADSTech USBX-707 USB IR Blaster;usbx;none;hw_usbx;adstech/lircd.conf.usbx-707;
 Apple Mac mini USB IR Receiver;macmini;none;hw_macmini;apple/lircd.conf.macmini;
 Asus DH USB Remote;asusdh;none;hw_asusdh;asus/lircd.conf.asusdh;
-ATI/NVidia/X10 I & II RF Remote;atiusb;lirc_dev lirc_atiusb;hw_default;atiusb/lircd.conf.atiusb;
-ATI/NVidia/X10 RF Remote (userspace);atilibusb;none;hw_atilibusb;atiusb/lircd.conf.atilibusb;
+ATI/NVidia X10 RF (kernel);atiusb;lirc_dev lirc_atiusb;hw_default;atiusb/lircd.conf.atiusb;
+ATI/NVidia X10 RF (userspace);atilibusb;none;hw_atilibusb;atiusb/lircd.conf.atilibusb;
+Snapstream X10 RF;atiusb;lirc_dev lirc_atiusb;hw_default;atiusb/lircd.conf.atiusb;
 Creative USB IR Receiver (SB0540);sb0540;none;hw_sb0540;creative/lircd.conf.alsa_usb;
 COMMANDIR USB Transceiver;cmdir;lirc_dev commandir lirc_cmdir;hw_default;;
 Dign HV5 HTPC IR/VFD Module;sasem;lirc_dev lirc_sasem;hw_default;sasem/lircd.conf.sasem;
diff -urNad lirc-0.8.3~pre1~/remotes/atiusb/lircd.conf.atilibusb lirc-0.8.3~pre1/remotes/atiusb/lircd.conf.atilibusb
--- lirc-0.8.3~pre1~/remotes/atiusb/lircd.conf.atilibusb	2007-09-14 15:45:18.000000000 -0400
+++ lirc-0.8.3~pre1/remotes/atiusb/lircd.conf.atilibusb	2007-10-30 15:48:16.000000000 -0400
@@ -248,3 +248,199 @@
       end codes
 
 end remote
+
+#
+# contributed by Michael Haas
+#
+# brand: ATI "RF Remote Control" (first generation)
+# model no. of remote control: Part #: 5000023600
+# FCC ID: B4SUR84A
+# picture: http://www.mythtv.org/wiki/index.php/Image:Remotewonder1.jpg
+# receiver: http://www.mythtv.org/wiki/index.php/Image:Remotewonder2.jpg
+#
+# I created this new config file because the one for the ati remote wonder II
+# did not work for me.
+
+begin remote
+
+  name ati_remote_wonder_rf
+  bits           40
+  eps            30
+  aeps          100
+
+  one             0     0
+  zero            0     0
+  gap          147991
+  toggle_bit	0
+
+      begin codes
+          0                        0x14DC170000
+          1                        0x14D20D0000
+          2                        0x14D30E0000
+          3                        0x14D40F0000
+          4                        0x14D5100000
+          5                        0x14D6110000
+          6                        0x14D7120000
+          7                        0x14D8130000
+          8                        0x14D9140000
+          9                        0x14DA150000
+          volup                    0x14CD080000
+          voldown                  0x14CE090000
+          mute                     0x14CF0A0000
+          chup                     0x14D00B0000
+          chdown                   0x14D10C0000
+          mouse_up                 0x1437720000
+          mouse_down               0x1438730000
+          mouse_left               0x1435700000
+          mouse_right              0x1436710000
+          mouse_up_right           0x143A750000
+          mouse_down_right         0x143B760000
+          mouse_up_left            0x1439740000
+          mouse_down_left          0x143C770000
+          right_click              0x14417C0000
+          left_click               0x143D780000
+          hand                     0x14CC070000
+          book                     0x14CB060000
+          ?                        0x14CA050000
+          dvd                      0x14C9040000
+          tv                       0x14C8030000
+          a                        0x14C5000000
+          b                        0x14C6010000
+          power                    0x14C7020000
+          stop                     0x14ED280000
+          pause                    0x14EE290000
+          fforward                 0x14EB260000
+          rec                      0x14EC270000
+          rewind                   0x14E9240000
+          play                     0x14EA250000
+          d                        0x14E01B0000
+          c                        0x14DE190000
+          e                        0x14E6210000
+          f                        0x14E8230000
+          right                    0x14E41F0000
+          left                     0x14E21D0000
+          up                       0x14DF1A0000
+          down                     0x14E7220000
+          ok                       0x14E31E0000
+          timer                    0x14E11C0000
+          shrink_resize            0x14E5200000
+          menu                     0x14DB160000
+          check                    0x14DD180000
+      end codes
+
+end remote
+
+begin remote
+
+name Snapstream Firefly
+bits 40
+eps 30
+aeps 100
+
+one 0 0
+zero 0 0
+gap 219964
+toggle_bit 0
+
+
+begin codes
+MAXI 0x0000001481AC0000
+MAXI 0x00000014012C0000
+CLOSE 0x00000014D7020000
+CLOSE 0x0000001457820000
+1 0x00000014628D0000
+1 0x00000014E20D0000
+2 0x00000014E30E0000
+2 0x00000014638E0000
+3 0x00000014648F0000
+3 0x00000014E40F0000
+4 0x00000014E5100000
+4 0x0000001465900000
+5 0x0000001466910000
+5 0x00000014E6110000
+6 0x00000014E7120000
+6 0x0000001467920000
+7 0x0000001468930000
+7 0x00000014E8130000
+8 0x00000014E9140000
+8 0x0000001469940000
+9 0x000000146A950000
+9 0x00000014EA150000
+0 0x00000014EC170000
+0 0x000000146C970000
+BACK 0x000000146B960000
+BACK 0x00000014EB160000
+ENT 0x00000014ED180000
+ENT 0x000000146D980000
+VOL+ 0x000000145E890000
+VOL+ 0x00000014DE090000
+VOL- 0x000000145D880000
+VOL- 0x00000014DD080000
+MUTE 0x000000145F8A0000
+MUTE 0x00000014DF0A0000
+FIREFLY 0x0000001455800000
+FIREFLY 0x00000014D5000000
+CH+ 0x00000014608B0000
+CH+ 0x00000014E00B0000
+CH- 0x00000014618C0000
+CH- 0x00000014E10C0000
+INFO 0x0000001483AE0000
+INFO 0x00000014032E0000
+OPTION 0x0000001484AF0000
+OPTION 0x00000014042F0000
+UP 0x000000146F9A0000
+UP 0x00000014EF1A0000
+LEFT 0x00000014729D0000
+LEFT 0x00000014F21D0000
+DOWN 0x0000001477A20000
+DOWN 0x00000014F7220000
+RIGHT 0x00000014749F0000
+RIGHT 0x00000014F41F0000
+OK 0x00000014739E0000
+OK 0x00000014F31E0000
+MENU 0x00000014719C0000
+MENU 0x00000014F11C0000
+EXIT 0x0000001475A00000
+EXIT 0x00000014F5200000
+REC 0x00000014FC270000
+REC 0x000000147CA70000
+PLAY 0x00000014FA250000
+PLAY 0x000000147AA50000
+STOP 0x00000014FD280000
+STOP 0x000000147DA80000
+REW 0x00000014F9240000
+REW 0x0000001479A40000
+FWD 0x00000014FB260000
+FWD 0x000000147BA60000
+PREV 0x00000014002B0000
+PREV 0x0000001480AB0000
+PAUSE 0x00000014FE290000
+PAUSE 0x000000147EA90000
+NEXT 0x00000014FF2A0000
+NEXT 0x000000147FAA0000
+MUSIC 0x00000014DB060000
+MUSIC 0x000000145B860000
+PHOTOS 0x00000014DA050000
+PHOTOS 0x000000145A850000
+DVD 0x00000014D9040000
+DVD 0x0000001459840000
+TV 0x00000014D8030000
+TV 0x0000001458830000
+VIDEO 0x00000014DC070000
+VIDEO 0x000000145C870000
+HELP 0x00000014D6010000
+HELP 0x0000001456810000
+MOUSE 0x00000014022D0000
+MOUSE 0x0000001482AD0000
+A 0x00000014EE190000
+A 0x000000146E990000
+B 0x00000014F01B0000
+B 0x00000014709B0000
+C 0x00000014F6210000
+C 0x0000001476A10000
+D 0x00000014F8230000
+D 0x0000001478A30000
+
+end codes
+
+end remote
diff -urNad lirc-0.8.3~pre1~/remotes/atiusb/lircd.conf.atiusb lirc-0.8.3~pre1/remotes/atiusb/lircd.conf.atiusb
--- lirc-0.8.3~pre1~/remotes/atiusb/lircd.conf.atiusb	2007-09-01 15:27:28.000000000 -0400
+++ lirc-0.8.3~pre1/remotes/atiusb/lircd.conf.atiusb	2007-10-30 15:48:16.000000000 -0400
@@ -191,7 +191,7 @@
 # this config file was automatically generated
 # using lirc-0.7.0-CVS(atiusb) on Sat May 15 10:44:51 2004
 #
-# contributed by Jurgen Kramer 
+# contributed by Jurgen Kramer
 #
 # brand: ATI
 # model no. of remote control: 5000023600
@@ -365,7 +365,7 @@
 # matching MythTV configuration file at http://wendy.seltzer.org/mythtv/lircrc
 #
 #
-# brand:	X-10 Lola                       
+# brand:	X-10 Lola
 # model no. of remote control:  UR89A
 # devices being controlled by this remote: mythtv
 #
@@ -379,7 +379,7 @@
 
   one             0     0
   zero            0     0
-  
+
   pre_data_bits   8
   pre_data       0x14
   post_data_bits  16
@@ -525,10 +525,10 @@
 # contributed by Martin Tomasek <mtd@centrum.cz>
 #
 # brand: ATI Remote Wonder
-# model no. of remote control: 
+# model no. of remote control:
 # devices being controlled by this remote: ATI USB Receiver (X10).
 #
-# * remote is set to channel 1, driver reports channel 16. don't know 
+# * remote is set to channel 1, driver reports channel 16. don't know
 # which is true.
 
 #
@@ -615,7 +615,7 @@
 # contributed by Phil Speights
 #
 # brand: nvidia
-# model no. of remote control: UR88A 
+# model no. of remote control: UR88A
 # devices being controlled by this remote: nvidia personal cinema
 #
 
@@ -875,10 +875,10 @@
 # this config file was automatically generated
 # using lirc-0.7.0(any) on Sun Jan 16 14:58:13 2005
 #
-# contributed by 
+# contributed by
 #
 # brand:                       Niveus
-# model no. of remote control: 
+# model no. of remote control:
 # devices being controlled by this remote: Niveus X10
 #
 
@@ -1024,7 +1024,7 @@
         Info            0x84af
         Menu            0x6e99
         left            0x729d
-        up_0            0x4d78  
+        up_0            0x4d78
         up_1            0x4e79
         up_2            0x4f7a
         up_3            0x507b
@@ -1064,7 +1064,7 @@
         Record          0x7ca7
         FastForward     0x7ba6
 
-    end codes  
+    end codes
 end remote
 
 
@@ -1076,7 +1076,7 @@
 #
 # brand: MEDION X10 RF
 # model no. of remote control: remote P/N:20014752, receiver P/N:20014751
-# devices being controlled by this remote: 
+# devices being controlled by this remote:
 #
 
 begin remote
@@ -1387,7 +1387,7 @@
 # this config file was automatically generated
 # using lirc-0.8.0(atiusb) on Mon Dec 11 20:45:22 2006
 #
-# contributed by Wim Lemmers 
+# contributed by Wim Lemmers
 #
 # brand:                       Medion
 # model no. of remote control: OR24E RF MCE Remote Control
@@ -1469,7 +1469,7 @@
 #
 # contributed by Harald Wild - www.wild-technology.de
 #
-# brand: Q-Sonic Master Remote PC / TV 
+# brand: Q-Sonic Master Remote PC / TV
 # model no. of remote control: Art.Nr. PE-9999
 # frequency: 433.92 MHz
 # distributor Germany: PEARL AGENCY - www.pearl.de
@@ -1703,9 +1703,9 @@
 # this config file was automatically generated
 # using lirc-0.8.2-CVS(atilibusb) on Fri Jun  8 13:19:29 2007
 #
-# contributed by 
+# contributed by
 #
-# brand:                       
+# brand:
 # model no. of remote control: ATI/X10 RF USB Remote Control v. 2.2.1
 # devices being controlled by this remote:
 #
@@ -1940,4 +1940,118 @@
 
 end remote
 
+begin remote
+
+name Snapstream Firefly
+bits 40
+eps 30
+aeps 100
+
+one 0 0
+zero 0 0
+gap 219964
+toggle_bit 0
+
+
+begin codes
+MAXI 0x0000001481AC0000
+MAXI 0x00000014012C0000
+CLOSE 0x00000014D7020000
+CLOSE 0x0000001457820000
+1 0x00000014628D0000
+1 0x00000014E20D0000
+2 0x00000014E30E0000
+2 0x00000014638E0000
+3 0x00000014648F0000
+3 0x00000014E40F0000
+4 0x00000014E5100000
+4 0x0000001465900000
+5 0x0000001466910000
+5 0x00000014E6110000
+6 0x00000014E7120000
+6 0x0000001467920000
+7 0x0000001468930000
+7 0x00000014E8130000
+8 0x00000014E9140000
+8 0x0000001469940000
+9 0x000000146A950000
+9 0x00000014EA150000
+0 0x00000014EC170000
+0 0x000000146C970000
+BACK 0x000000146B960000
+BACK 0x00000014EB160000
+ENT 0x00000014ED180000
+ENT 0x000000146D980000
+VOL+ 0x000000145E890000
+VOL+ 0x00000014DE090000
+VOL- 0x000000145D880000
+VOL- 0x00000014DD080000
+MUTE 0x000000145F8A0000
+MUTE 0x00000014DF0A0000
+FIREFLY 0x0000001455800000
+FIREFLY 0x00000014D5000000
+CH+ 0x00000014608B0000
+CH+ 0x00000014E00B0000
+CH- 0x00000014618C0000
+CH- 0x00000014E10C0000
+INFO 0x0000001483AE0000
+INFO 0x00000014032E0000
+OPTION 0x0000001484AF0000
+OPTION 0x00000014042F0000
+UP 0x000000146F9A0000
+UP 0x00000014EF1A0000
+LEFT 0x00000014729D0000
+LEFT 0x00000014F21D0000
+DOWN 0x0000001477A20000
+DOWN 0x00000014F7220000
+RIGHT 0x00000014749F0000
+RIGHT 0x00000014F41F0000
+OK 0x00000014739E0000
+OK 0x00000014F31E0000
+MENU 0x00000014719C0000
+MENU 0x00000014F11C0000
+EXIT 0x0000001475A00000
+EXIT 0x00000014F5200000
+REC 0x00000014FC270000
+REC 0x000000147CA70000
+PLAY 0x00000014FA250000
+PLAY 0x000000147AA50000
+STOP 0x00000014FD280000
+STOP 0x000000147DA80000
+REW 0x00000014F9240000
+REW 0x0000001479A40000
+FWD 0x00000014FB260000
+FWD 0x000000147BA60000
+PREV 0x00000014002B0000
+PREV 0x0000001480AB0000
+PAUSE 0x00000014FE290000
+PAUSE 0x000000147EA90000
+NEXT 0x00000014FF2A0000
+NEXT 0x000000147FAA0000
+MUSIC 0x00000014DB060000
+MUSIC 0x000000145B860000
+PHOTOS 0x00000014DA050000
+PHOTOS 0x000000145A850000
+DVD 0x00000014D9040000
+DVD 0x0000001459840000
+TV 0x00000014D8030000
+TV 0x0000001458830000
+VIDEO 0x00000014DC070000
+VIDEO 0x000000145C870000
+HELP 0x00000014D6010000
+HELP 0x0000001456810000
+MOUSE 0x00000014022D0000
+MOUSE 0x0000001482AD0000
+A 0x00000014EE190000
+A 0x000000146E990000
+B 0x00000014F01B0000
+B 0x00000014709B0000
+C 0x00000014F6210000
+C 0x0000001476A10000
+D 0x00000014F8230000
+D 0x0000001478A30000
+
+end codes
+
+end remote
 
